src/windows/minidump_writer.rs

```diff
@@ -0,0 +1,279 @@
+use crate::windows::errors::Error;
+use minidump_common::format::{BreakpadInfoValid, MINIDUMP_BREAKPAD_INFO, MINIDUMP_STREAM_TYPE};
+use scroll::Pwrite;
+use std::{ffi::c_void, os::windows::io::AsRawHandle};
+pub use windows_sys::Win32::Foundation::HANDLE;
+use windows_sys::Win32::{
+    Foundation::{CloseHandle, ERROR_SUCCESS, STATUS_INVALID_HANDLE},
+    System::{
+        ApplicationVerifier as av,
+        Diagnostics::Debug as md,
+        Threading::{GetCurrentThreadId, OpenProcess},
+    },
+};
+
+// For some reason this is defined in SystemServices which is massive and not worth bringing in for ONE constant
+const GENERIC_ALL: u32 = 268435456;
+
+pub struct MinidumpWriter {
+    /// The crash context as captured by an exception handler
+    crash_context: crash_context::CrashContext,
+    /// Handle to the crashing process, which could be ourselves
+    crashing_process: HANDLE,
+    /// The pid of the crashing process.
+    crashing_pid: u32,
+    /// The `EXCEPTION_POINTERS` contained in crash context is a pointer into the
+    /// memory of the process that crashed, as it contains an `EXCEPTION_RECORD`
+    /// record which is an internally linked list, so in the case that we are
+    /// dumping a process other than the current one, we need to tell
+    /// MiniDumpWriteDump that the pointers come from an external process so that
+    /// it can use eg ReadProcessMemory to get the contextual information from
+    /// the crash, rather than from the current process
+    is_external_process: bool,
+}
+
+impl MinidumpWriter {
+    /// Creates a minidump writer for a crash that occurred in an external process.
+    pub fn external_process(
+        crash_context: crash_context::CrashContext,
+        pid: u32,
+    ) -> Result<Self, Error> {
+        let crashing_process = unsafe {
+            OpenProcess(
+                GENERIC_ALL, // desired access
```

Comment:
Is there a reason for preferring this instead of `PROCESS_ALL_ACCESS`?

It's interesting that this value works, as I don't see it mentioned in the [MSDN docs for OpenProcess](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-openprocess). `PROCESS_ALL_ACCESS` also has the advantage of being in `Win32_System_Threading`, so you wouldn't have to define the special constant above

Reply:
This was just the access that the original Breakpad code was using and it just worked so actually didn't even look at the documentation for this function, but can try changing it.

Reply:
Yup seems to work just fine.